sunscreen_tfhe/gpu_src/src/math/fft/negacyclic.cuh

```diff
@@ -47,32 +43,30 @@ __device__ void twisted_fft(
 
 /// Compute a twisted IFFT resulting in real negacyclic output `s_output`.
 /// This result is still in floating point and needs to be modulo reduced.
-template <typename Complex>
+template <typename T>
 __device__ void twisted_ifft(
-    Complex * __restrict__ s_input,
-    typename ScalarOf<Complex>::Ty * __restrict__ s_output,
+    Complex<T> * __restrict__ s_input,
+    T * __restrict__ s_output,
     uint32_t n)
 {
-    using S = typename ScalarOf<Complex>::Ty;
     uint32_t n_div_2 = n / 2;
 
     // Perform an n/2 IFFT.
     ifft(s_input, n_div_2);
 
-    S n_inv = 1.0 / (S)n_div_2;
+    T n_inv = 1.0 / (T)n_div_2;
 
     // Twist the inputs so we can use an N-point FFT for negacyclic
     // convolution
     for (uint32_t i = threadIdx.x; i < n_div_2; i += blockDim.x)
     {
-        Complex twist_inv;
-        sincos(-PI * (S)i / (S)n, &twist_inv.y, &twist_inv.x);
+        Complex<T> twist_inv;
+        sincos(-PI * (T)i / (T)n, &twist_inv.im(), &twist_inv.re());
```

Comment:
Add a TODO for calling the twiddles method in fft.cuh so we can use LUT or sincospi as per #defines.